package/MDAnalysis/auxiliary/EDR.py

```diff
@@ -0,0 +1,358 @@
+Only the dictionary syntax will work in that case.
+
+
+Adding multiple or all terms at the same time is possible as such::
+
+    # add list of values
+    u.trajectory.add_auxiliary(["Bond", "Angle"], aux)
+    # add list of values, provide an `auxterm`
+    u.trajectory.add_auxiliary(["epot", "ekin"], aux, ["Potential",
+                                                       "Kinetic En."]
+    # add all values
+    u.trajectory.add_auxiliary("*", aux)
+
+
+
+The EDRReader can also provide the data independently of trajectories. This is
+useful, for example, for plotting. The data for all terms, a list of terms, or
+a single term can be returned in dictionary form. "Time" is always returned in
+this dictionary to make plotting easier::
+
+    # extract temperature data from EDR file
+    temp = aux.return_data("Temperature")
+    plt.plot(temp["Time"], temp["Temperature"])
+
+    # extract list of terms:
+    some_terms = aux.return_data(["Potential", "Kinetic En.", "Box-X"])
+
+    # extract all terms:
+    all_terms = aux.return_data()
+    # alternatively
+    all_terms = aux.return_data("*")
+
+.. _EDR: https://manual.gromacs.org/current/reference-manual/file-formats.html#edr
+.. _XDR: https://datatracker.ietf.org/doc/html/rfc1014
+.. _pyedr: https://github.com/mdanalysis/panedr
+.. _GROMACS: https://github.com/gromacs/gromacs/blob/main/src/gromacs/fileio/enxio.cpp
+
+
+.. autoclass:: EDRReader
+   :members:
+
+
+"""
+from pathlib import Path
+from . import base
+import pyedr
+import numpy as np
+from typing import Optional, Union, Dict, List
+
+
+class EDRStep(base.AuxStep):
+    """:class:`AuxStep` class for .edr file format.
+
+    Extends the base AuxStep class to allow selection of time and
+    data-of-interest fields (by dictionary key) from the full set of data read
+    each step.
+
+    Parameters
+    ----------
+    time_selector : str, optional
+        Name of the dictionary key that links to the time values (assumed to
+        be in ps). Default value is "Time"
+    data_selector : str | list of str | None, optional
+        List of dictionary keys linking to data of interest in the EDR file to
+        be stored in ``data``. Default value is ``None``.
+    **kwargs
+        Other AuxStep options.
+
+    See Also
+    --------
+    :class:`~MDAnalysis.auxiliary.base.AuxStep`
+    """
+
+    def __init__(self, time_selector: str = "Time",
+                 data_selector: Optional[str] = None, **kwargs):
+        super(EDRStep, self).__init__(time_selector=time_selector,
+                                      data_selector=data_selector,
+                                      **kwargs)
+
+    def _select_time(self, key: str) -> np.float64:
+        """'Time' is one of the entries in the dict returned by pyedr.
+        The base AuxStep Class uses the time_selector 'Time' to return the
+        time value of each step."""
+        return self._select_data(key)
```

Comment:
I'm all for small methods, but is there an advantage to having `_select_data` separate here, especially if it's not a staticmethod?

Reply:
Both `_select_time` and `_select_data` need to be defined for `AuxReaders` so the base class works. I could just have `_select_time` return the value from the dictionary itself without calling `_select_data`, but then I'd also need to duplicate the try-except which I find less clean